### PR TITLE
CLI command to create extensions for dynamic contracts

### PR DESCRIPTION
**Pull request by nkrishang**
Adds command:
```bash
thirdweb create extension
```

which creates a boilerplate extension for dynamic smart contracts (e.g. for `TestExtension`):
```Solidity
// SPDX-License-Identifier: UNLICENSED
pragma solidity ^0.8.0;

library TestExtensionStorage {
  bytes32 public constant TEST_EXTENSION_STORAGE_POSITION = keccak256("test.extension.storage");

  // Define your contract's state inside the Data struct.
  struct Data {
    // Example: a uint called "number"
    uint256 number;
  }

  /**
    *  Access your state variables using the following pattern (example):
    *  TestExtensionStorage.Data storage data = TestExtensionStorage.getStorage();
  **/
  function getStorage() internal pure returns (Data storage data) {
    bytes32 position = TEST_EXTENSION_STORAGE_POSITION;
    assembly {
      data.slot := position
    }
  }
}
  
contract TestExtension {

  function readContractState() external view returns (uint256) {
    TestExtensionStorage.Data storage data = TestExtensionStorage.getStorage();
    return data.number;
  }

  function writeContractState(uint256 x) external {
    TestExtensionStorage.Data storage data = TestExtensionStorage.getStorage();
    data.number = x;
  }
}
```

### CODE REVIEW

File: packages/cli/src/cli/index.ts [hunk 1]
```diff
@@ -214,6 +214,7 @@ const main = async () => {
     .option("--js, --javascript", "Initialize as a JavaScript project.")
     .option("--forge", "Initialize as a Forge project.")
     .option("--hardhat", "Initialize as a Hardhat project.")
+    .option("--extension", "Create a smart contract extension.")
```
Comment (joaquim-verges) [inline packages/cli/src/cli/index.ts:217]: not for right now, but wonder if we should make this a 3rd option from `npx create app/contract/extension`

File: packages/cli/src/create/helpers/create-contract.ts [hunk 2]
```diff
@@ -99,6 +103,23 @@ export async function createContractProject({
...
+
+      // Set the contents of the Contract.sol file to the extension contract storage library boilerplate
+      let contractFile = "";
+      contractFile = path.join(root, contractName);
+
+      // Write the extension storage library to the MyContract.sol file
+      await writeFile(contractFile, extensionBoilerplate);
+
+      console.log(
+        `${chalk.green("Success!")} Create your ${contractName} extension at ${contractPath}`,
```
Comment (joaquim-verges) [inline packages/cli/src/create/helpers/create-contract.ts:119]: Create -> Created

File: packages/cli/src/create/helpers/create-contract.ts [hunk 3]
```diff
@@ -99,6 +103,23 @@ export async function createContractProject({
         `${chalk.green("Success!")} Created ${contractName} at ${contractPath}`,
       );
     }
+
+    if(!baseContract && createExtension) {
```
Comment (joaquim-verges) [inline packages/cli/src/create/helpers/create-contract.ts:107]: wonder if ppl are gonna create project + extension in one go 🤔 guess it doesnt hurt to have it

File: packages/cli/src/create/helpers/create-contract.ts [hunk 4]
```diff
@@ -164,6 +185,27 @@ export async function createContractProject({
...
+  
+        // Set the contents of the Contract.sol file to the extension contract storage library boilerplate
+        let contractFile = "";
+        contractFile = path.join(root, contractName);
+        if (framework === "hardhat") {
+          fs.unlinkSync(path.join(root, "contracts", "Contract.sol"));
+          contractFile = path.join(root, "contracts", contractName);
+        }
+        if (framework === "forge") {
+          fs.unlinkSync(path.join(root, "src", "Contract.sol"));
```
Comment (joaquim-verges) [inline packages/cli/src/create/helpers/create-contract.ts:201]: Does the file have to be called `Contract.sol` ? can't we use the extension name entered?

File: yarn.lock [hunk 5]
```diff
@@ -24,11 +24,6 @@
   resolved "https://registry.yarnpkg.com/@apocentre/alias-sampling/-/alias-sampling-0.5.3.tgz#897ff181b48ad7b2bcb4ecf29400214888244f08"
   integrity sha512-7UDWIIF9hIeJqfKXkNIzkVandlwLf1FWTSdrb9iXvOP8oF544JRXQjCbiTmCv2c9n44n/FIWtehhBfNuAx2CZA==
 
-"@assemblyscript/loader@^0.9.4":
```
Comment (joaquim-verges) [inline yarn.lock:27]: oh wow all these dependencies were uneeded ? weird

File: packages/cli/src/cli/index.ts (hunk 1 above)
Comment (jnsdls) [reply packages/cli/src/cli/index.ts:217]: yes please just make it `extension` namespace

File: packages/cli/src/create/helpers/create-contract.ts (hunk 4 above)
Comment (nkrishang) [reply packages/cli/src/create/helpers/create-contract.ts:201]: We do use the extension name entered 👍🏽

**Pull request merged**